cou/utils/juju_utils.py

```diff
@@ -396,3 +397,101 @@ async def async_upgrade_charm(
         revision=revision,
         switch=switch,
     )
+
+
+# pylint: disable=too-few-public-methods
+class JujuWaiter:
+    """Enhanced version of wait_for_idle.
+
+    Usage:
+        JujuWaiter(model).wait(120)
+    """
+
+    DEFAULT_TIMEOUT: int = 3600
+    MODEL_IDLE_PERIOD: int = 30
+    JUJU_IDLE_CHECK_PERIOD: int = 40
+
+    class TimeoutException(Exception):
+        """Own timeout exception."""
+
+    def __init__(self, model: Model):
+        """Initialize.
+
+        :param model: model to wait
+        :type model: Model
+        """
+        self.model = model
+        self.model_name = self.model.info.name
+        self.timeout = timedelta(seconds=JujuWaiter.DEFAULT_TIMEOUT)
+        self.start_time = datetime.now()
+        self.log = logging.getLogger(self.__class__.__name__)
+
+    async def wait(self, timeout_seconds: Optional[int] = None) -> None:
```

Comment:
When we will need to use waiting without timeout? I think this is not good idea to wait forever.

Reply:
It is initialized in constructor with some sane default
`self.timeout = timedelta(seconds=JujuWaiter.DEFAULT_TIMEOUT)`

Reply:
This is not good at all, because we are hiding that there is default value. It should be `timeout: int = DEFAULT_TIMEOUT`

Reply:
changed as suggested.